src/contexts/states/viewStore.ts

```diff
@@ -0,0 +1,246 @@
+
+const VIEW_STORE_DEFAULT: ViewStoreValues = {
+    beginLineNumToLogEventNum: new Map<number, number>(),
+    isPrettified: false,
+    logData: "No file is open.",
+    numPages: 0,
+    pageNum: 0,
+};
+
+/**
+ * Returns a `PAGE_NUM` cursor based on a navigation action.
+ *
+ * @param navAction Action to navigate to a new page.
+ * @param currentPageNum
+ * @param numPages
+ * @return `PAGE_NUM` cursor.
+ */
+const getPageNumCursor = (
+    navAction: NavigationAction,
+    currentPageNum: number,
+    numPages: number
+): Nullable<CursorType> => {
+    let newPageNum: number;
+    let position: EVENT_POSITION_ON_PAGE;
+    switch (navAction.code) {
+        case ACTION_NAME.SPECIFIC_PAGE:
+            position = EVENT_POSITION_ON_PAGE.TOP;
+
+            // Clamp is to prevent someone from requesting non-existent page.
+            newPageNum = clamp(navAction.args.pageNum, 1, numPages);
+            break;
+        case ACTION_NAME.FIRST_PAGE:
+            position = EVENT_POSITION_ON_PAGE.TOP;
+            newPageNum = 1;
+            break;
+        case ACTION_NAME.PREV_PAGE:
+            position = EVENT_POSITION_ON_PAGE.BOTTOM;
+            newPageNum = clamp(currentPageNum - 1, 1, numPages);
+            break;
+        case ACTION_NAME.NEXT_PAGE:
+            position = EVENT_POSITION_ON_PAGE.TOP;
+            newPageNum = clamp(currentPageNum + 1, 1, numPages);
+            break;
+        case ACTION_NAME.LAST_PAGE:
+            position = EVENT_POSITION_ON_PAGE.BOTTOM;
+            newPageNum = numPages;
+            break;
+        default:
+            return null;
+    }
+
+    return {
+        code: CURSOR_CODE.PAGE_NUM,
+        args: {pageNum: newPageNum, eventPositionOnPage: position},
+    };
+};
+
+// eslint-disable-next-line max-lines-per-function
+const useViewStore = create<ViewState>((set, get) => ({
+    ...VIEW_STORE_DEFAULT,
+    filterLogs: (filter: LogLevelFilter) => {
+        const {setUiState} = useUiStore.getState();
+        setUiState(UI_STATE.FAST_LOADING);
+        const {logEventNum} = useContextStore.getState();
+
+        useLogFileManagerStore
+            .getState()
+            .logFileManagerProxy.setFilter(
+                {code: CURSOR_CODE.EVENT_NUM,
+                    args: {
+                        eventNum: 0 === logEventNum ?
+                            1 :
+                            logEventNum,
+                    }},
+                get().isPrettified,
+                filter
+            ).then((pageData: PageData) => {
+                useViewStore.getState().updatePageData(pageData);
+            })
+            .catch((e: unknown) => {
+                useContextStore.getState().postPopUp({
+                    level: LOG_LEVEL.ERROR,
+                    message: String(e),
+                    timeoutMillis: DO_NOT_TIMEOUT_VALUE,
+                    title: "Action failed",
+                });
+            });
+        useQueryStore.getState().startQuery();
```

Comment:
it's nice we're getting #168 easily fixed in this PR. let's make sure the PR title is update to reflect that